src/twisted/internet/base.py

```diff
@@ -1074,65 +1164,79 @@ def _stopThreadPool(self):
                     pass
             self._threadpoolStartupID = None
             self.threadpoolShutdownID = None
+            assert self.threadpool is not None
             self.threadpool.stop()
             self.threadpool = None
 
-        def getThreadPool(self):
+        def getThreadPool(self) -> ThreadPool:
             """
             See L{twisted.internet.interfaces.IReactorThreads.getThreadPool}.
             """
             if self.threadpool is None:
                 self._initThreadPool()
+                assert self.threadpool is not None
             return self.threadpool
 
-        def callInThread(self, _callable, *args, **kwargs):
+        def callInThread(
+            self, _callable: Callable[..., Any], *args: object, **kwargs: object
+        ) -> None:
             """
             See L{twisted.internet.interfaces.IReactorInThreads.callInThread}.
             """
             self.getThreadPool().callInThread(_callable, *args, **kwargs)
 
-        def suggestThreadPoolSize(self, size):
+        def suggestThreadPoolSize(self, size: int) -> None:
             """
             See L{twisted.internet.interfaces.IReactorThreads.suggestThreadPoolSize}.
             """
             self.getThreadPool().adjustPoolsize(maxthreads=size)
 
     else:
         # This is for signal handlers.
-        def callFromThread(self, f, *args, **kw):
+        def callFromThread(
+            self, f: Callable[..., Any], *args: object, **kwargs: object
+        ) -> None:
             assert callable(f), "%s is not callable" % (f,)
             # See comment in the other callFromThread implementation.
-            self.threadCallQueue.append((f, args, kw))
+            self.threadCallQueue.append((f, args, kwargs))
 
 
 if platform.supportsThreads():
     classImplements(ReactorBase, IReactorThreads)
 
 
 @implementer(IConnector)
-class BaseConnector:
-    """Basic implementation of connector.
+class BaseConnector(ABC):
```

Comment:
Making this an ABC so that I can define the type signature of `_makeTransport`.

There are other classes around here that could probably benefit from becoming ABCs.